src/m365/spo/commands/site/site-apppermission-list.ts

```diff
@@ -63,26 +65,47 @@ class SpoSiteAppPermissionListCommand extends GraphCommand {
     });
   }
 
-  private getTransposed(permissions: SitePermission[]): { appDisplayName: string; appId: string; permissionId: string }[] {
-    const transposed: { appDisplayName: string; appId: string; permissionId: string }[] = [];
+  private getApplicationPermission(permissionId: string): Promise<SitePermission> {
+    return new Promise<SitePermission>((resolve: (permissionInfo: SitePermission) => void, reject: (error: any) => void): void => {
+      const requestOptions: any = {
+        url: `${this.resource}/v1.0/sites/${this.siteId}/permissions/${permissionId}`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
+
+      request
+        .get<SitePermission>(requestOptions)
+        .then((res: SitePermission): void => {
+          resolve(res);
+        }, (err: any): void => {
+          reject(err);
+        });
+    });
+  }
```

Comment:
Since we're not processing the data retrieved from the API in any way, we can simplify the method to:

```ts
private getApplicationPermission(permissionId: string): Promise<SitePermission> {
    const requestOptions: any = {
      url: `${this.resource}/v1.0/sites/${this.siteId}/permissions/${permissionId}`,
      headers: {
        accept: 'application/json;odata.metadata=none'
      },
      responseType: 'json'
    };

    return request.get<SitePermission>(requestOptions);
  }
```